lib/Doctrine/ODM/PHPCR/UnitOfWork.php

```diff
@@ -3032,8 +3041,19 @@ public function registerDocument($document, $id)
         $oid = spl_object_hash($document);
         $this->documentIds[$oid] = $id;
         $this->identityMap[$id] = $document;
-        $this->setDocumentState($oid, self::STATE_MANAGED);
-
+        
+        // frozen nodes need another state so they are managed but not included for updates
+        $nodeType = '';
+        try {
+        	$node = $this->session->getNode($id);
+        	// In some of the unit tests $node will be null because of how they are mocked
+        	$nodeType = $node ? $node->getPrimaryNodeType()->getName() : '';
```

Comment:
Addressed - used your $frozen example - much neater and no exceptions for tests required